tfhe-zk-pok/src/proofs/pke.rs

```diff
@@ -1457,56 +1369,30 @@ mod tests {
 
     #[test]
     fn test_pke_w_padding_fail_verify() {
-        let d = 2048;
-        let k = 320;
-        let b_i = 512;
-        let q = 0;
-        let t = 1024;
-        let msbs_zero_padding_bit_count = 1;
-        let effective_cleartext_t = t >> msbs_zero_padding_bit_count;
+        let PkeTestParameters {
+            d,
+            k,
+            B,
+            q,
+            t,
+            msbs_zero_padding_bit_count,
+        } = PKEV1_TEST_PARAMS;
 
-        let delta = {
-            let q = if q == 0 { 1i128 << 64 } else { q as i128 };
-            (q / t as i128) as u64
-        };
+        let effective_cleartext_t = t >> msbs_zero_padding_bit_count;
 
         let rng = &mut StdRng::seed_from_u64(0);
 
-        let polymul_rev = |a: &[i64], b: &[i64]| -> Vec<i64> {
-            assert_eq!(a.len(), b.len());
-            let d = a.len();
-            let mut c = vec![0i64; d];
-
-            for i in 0..d {
-                for j in 0..d {
-                    if i + j < d {
-                        c[i + j] = c[i + j].wrapping_add(a[i].wrapping_mul(b[d - j - 1]));
-                    } else {
-                        c[i + j - d] = c[i + j - d].wrapping_sub(a[i].wrapping_mul(b[d - j - 1]));
-                    }
-                }
-            }
-
-            c
-        };
-
-        let a = (0..d).map(|_| rng.gen::<i64>()).collect::<Vec<_>>();
-        let s = (0..d)
-            .map(|_| (rng.gen::<u64>() % 2) as i64)
-            .collect::<Vec<_>>();
-        let e = (0..d)
-            .map(|_| (rng.gen::<u64>() % (2 * b_i)) as i64 - b_i as i64)
-            .collect::<Vec<_>>();
-        let e1 = (0..d)
-            .map(|_| (rng.gen::<u64>() % (2 * b_i)) as i64 - b_i as i64)
-            .collect::<Vec<_>>();
-        let e2 = (0..k)
-            .map(|_| (rng.gen::<u64>() % (2 * b_i)) as i64 - b_i as i64)
-            .collect::<Vec<_>>();
-
-        let r = (0..d)
-            .map(|_| (rng.gen::<u64>() % 2) as i64)
-            .collect::<Vec<_>>();
+        let PkeTestProofInputs {
+            a,
+            e1,
+            e2,
+            r,
+            m: _m,
+            b,
+            c1,
+            c2,
+            metadata,
+        } = PkeTestProofInputs::gen(rng, PKEV1_TEST_PARAMS);
 
         // Generate messages with padding set to fail verification
         let m = {
```

Comment:
this is not correct, here this m is equivalent to a "fake m" what we want for the padding bit test is :

generate a legit encryption but with m that has at least one slot with a padding bit set and with that proven data we will check that the verification fails, indicating we are able to properly detect when a padding bit is set

Reply:
You're right, good catch !